code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/GlobalServiceDeploymentPlanner.java

```diff
@@ -36,10 +35,7 @@ public GlobalServiceDeploymentPlanner(List<Service> services, List<DeploymentUni
     public List<DeploymentUnit> deployHealthyUnits() {
         if (this.healthyUnits.size() < hostIds.size()) {
             addMissingUnits();
-        } else if (healthyUnits.size() > hostIds.size()) {
```

Comment:
Per https://github.com/rancherio/rancher/issues/1114 , do not remove containers already running